launcher/ui/GuiUtil.cpp

```diff
@@ -49,11 +50,34 @@
 #include <DesktopServices.h>
 #include <BuildConfig.h>
 
-QString GuiUtil::uploadPaste(const QString &text, QWidget *parentWidget)
+QString GuiUtil::uploadPaste(const QString &name, const QString &text, QWidget *parentWidget)
 {
     ProgressDialog dialog(parentWidget);
     auto pasteTypeSetting = static_cast<PasteUpload::PasteType>(APPLICATION->settings()->get("PastebinType").toInt());
     auto pasteCustomAPIBaseSetting = APPLICATION->settings()->get("PastebinCustomAPIBase").toString();
+
+    {
+        QUrl baseUrl;
+        if (pasteCustomAPIBaseSetting.isEmpty())
+            baseUrl = PasteUpload::PasteTypes[pasteTypeSetting].defaultBase;
+        else
+            baseUrl = pasteCustomAPIBaseSetting;
+
+        if (baseUrl.isValid()) {
+            auto response = CustomMessageBox::selectable(parentWidget, QObject::tr("Confirm Upload"),
+                                                         QObject::tr("About to upload: %1\n"
+                                                                     "Uploading to: %2\n"
+                                                                     "You should double-check for personal information.\n\n"
+                                                                     "Are you sure?")
+                                                             .arg(name, baseUrl.host()),
+                                                         QMessageBox::Warning, QMessageBox::Yes | QMessageBox::No, QMessageBox::No)
+                                ->exec();
+
+            if (response != QMessageBox::Yes)
+                return "canceled";
```

Comment:
Maybe return an **empty** string instead

```suggestion
                return "";
```

Reply:
An empty string (QString()) is already returned on failure.

Reply:
Empty string (`""`) and null-string (`QString()`) are distinct things!

.isEmpty will be true for both of them, while .isNull will only be true for `QString()`

Reply:
I don't think we should rely on that distinction though, it's very confusing and non-intuitive, especially if new contributors try to mess with that code

Reply:
> Empty string (`""`) and null-string (`QString()`) are distinct things!
> 
> .isEmpty will be true for both of them, while .isNull will only be true for `QString()`

Ah, I thought I could be missing something. I'm pretty new to Qt!

Reply:
> I don't think we should rely on that distinction though, it's very confusing and non-intuitive, 

But I think it's weird that this function would return anything that doesn't look empty but also isn't a URL.

Maybe std::optional instead then?

Reply:
either that, or a `std::pair` with the QString, and a return code (or better yet, a value from an enum class)

Reply:
Unless you *strongly* disagree, I think I'll go with Scrumplex's suggestion. Considering it's a null string rather than an empty one that's logical enough.

Reply:
I guess it's fine, it's just still confusing that there's now two "pseudo-failure" states: a real failure, that returns an empty optional, and a cancel, that returns an empty string. The function isn't important enough for me to double down on it though, so do whatever you see fitting :P

Reply:
I really do see what you mean but I'm not sure a custom enum and struct or pair would be worth it. Also I think I did it back-to-front with failure being null-string, and cancel being empty optional.

Reply:
yea do whatever you think works best for now, i just can't guarantee i won't change it to a pair in a future refactor xD /lh

Reply:
> yea do whatever you think works best for now, i just can't guarantee i won't change it to a pair in a future refactor xD /lh

Haha, I'm fine with that!